hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/client/DefaultCertificateClient.java

```diff
@@ -65,24 +79,61 @@
  */
 public abstract class DefaultCertificateClient implements CertificateClient {
 
+  private static final String CERT_FILE_NAME_FORMAT = "%s.crt";
   private final Logger logger;
   private final SecurityConfig securityConfig;
   private final KeyCodec keyCodec;
   private PrivateKey privateKey;
   private PublicKey publicKey;
   private X509Certificate x509Certificate;
+  private Map<String, X509Certificate> certificateMap;
 
 
   DefaultCertificateClient(SecurityConfig securityConfig, Logger log) {
     Objects.requireNonNull(securityConfig);
     this.securityConfig = securityConfig;
     keyCodec = new KeyCodec(securityConfig);
     this.logger = log;
+    this.certificateMap = new ConcurrentHashMap<>();
+
+    loadAllCertificates();
   }
 
   /**
-   * Returns the private key of the specified component if it exists on the
-   * local system.
+   * Load all certificates from configured location.
+   * */
+  private void loadAllCertificates() {
+    // See if certs directory exists in file system.
+    Path certPath = securityConfig.getCertificateLocation();
+    if (Files.exists(certPath) && Files.isDirectory(certPath) &&
+        certPath.toFile().listFiles() != null) {
+      getLogger().info("Loading certificate from location:{}.",
+          certPath);
+
+      for (File file : certPath.toFile().listFiles()) {
```

Comment:
There are two issues here:
1. 2nd listFIles() call can be avoided on line: 113.
2. This seems load only the local cert not those downloaded from scm because they are saved under subdirectories (certSerialID)

Reply:
2nd listFIles() call can be avoided on line: 113.
removed other reference.
This seems load only the local cert not those downloaded from scm because they are saved under subdirectories (certSerialID)
Patch stores all certificate in same certs dir.